lib/SftpClient.js

```diff
@@ -0,0 +1,86 @@
+const SFTP = require('sftp-promises')
+const FileParser = require('ftp/lib/parser')
+const { Duplex } = require('stream')
+
+class SftpClient {
+  constructor ({ host, port = 21, user, password }) {
+    this.host = host
+    this.port = port
+    this.user = user
+    this.password = password
+    this.client = new SFTP()
+    this.session = null
+  }
+
+  async connect () {
+    this.session = await this.client.session({
+      host: this.host,
+      port: this.port,
+      username: this.user,
+      password: this.password
+    })
+
+    return this.session
+  }
+
+  async disconnect () {
+    if (this.session) {
+      return this.session.end()
+    }
+  }
+
+  async list (path) {
+    return this.client.ls(path, this.session).then(result => {
+      return result.entries.map(entry => FileParser.parseListEntry(entry.longname))
+    })
+  }
+
+  async move (source, target) {
+    return this.client.mv(source, target, this.session)
+  }
+
+  async read (path) {
+    // Using `getStream` because `createReadStream` causes issues on Linux
+    const stream = new Pipe()
```

Comment:
Did you try [PassThrough
](https://nodejs.org/api/stream.html#stream_class_stream_passthrough)? It's also included in the `readable-stream` package. If for some reason it doesn't work, add a comment why and put the `Pipe` class into a separate file. When that's done, the PR can be merged.

Reply:
I missed PassThrough when reading the docs. It looks like exactly what I was looking for, nice!
I'll change it and merge.